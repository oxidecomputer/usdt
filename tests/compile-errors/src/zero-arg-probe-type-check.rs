//! Test that a zero-argument probe is correctly type-checked

// Copyright 2021 Oxide Computer Company

#![feature(asm)]
#![cfg_attr(target_os = "macos", feature(asm_sym))]

#[usdt::provider]
mod my_provider {
    fn my_probe() {}
}

fn main() {
    my_provider::my_probe!(|| "This should fail");
}
