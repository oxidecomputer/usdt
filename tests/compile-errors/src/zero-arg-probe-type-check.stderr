error[E0308]: mismatched types
  --> $DIR/zero-arg-probe-type-check.rs:7:1
   |
7  | #[usdt::provider]
   | ^^^^^^^^^^^^^^^^^
   | |
   | expected `()`, found `&str`
   | expected due to this
...
13 |     my_provider_my_probe!(|| "This should fail");
   |     -------------------------------------------- in this macro invocation
   |
   = note: this error originates in the macro `my_provider_my_probe` (in Nightly builds, run with -Z macro-backtrace for more info)
